### PR TITLE
102 c testing in flink

### PR DESCRIPTION
**Pull request by grittaweisheit**
This PR addresses #102 

It ...
- uses testcontainers for the integration tests 
    - weather-location-correlation
    - live-planned-train-correlation
    - data-to-database-map
- fixes minor issues in the tested functionalities

### CODE REVIEW

File: core/src/main/java/org/bptlab/cepta/operators/DataToDatabase.java [hunk 1]
```diff
@@ -15,25 +15,28 @@
...
-  static final String PASSWORD = "password";
-  static final Integer PORT = 5432;
   private String table_name;
+  private PostgresConfig postgresConfig = new PostgresConfig();
 
-  public DataToDatabase(String table){
+  public DataToDatabase(String table, PostgresConfig postgresConfig){
     this.table_name = table;
+    this.postgresConfig = postgresConfig;
+    System.out.print("KONSTRUKTOR");
```
Comment (leopetter) [inline core/src/main/java/org/bptlab/cepta/operators/DataToDatabase.java]: Wurde das zum debuggen verwendet?

File: core/src/main/java/org/bptlab/cepta/operators/DataToDatabase.java [hunk 2]
```diff
@@ -15,25 +15,28 @@
...
   }
 
   @Override
   public T map(T dataSet) throws Exception {
     insert(dataSet);
+    System.out.print("MAP");
```
Comment (leopetter) [inline core/src/main/java/org/bptlab/cepta/operators/DataToDatabase.java]: Wurde das ebenso zum debuggen verwendet?

File: core/src/main/java/org/bptlab/cepta/operators/WeatherLocationCorrelationFunction.java [hunk 3]
```diff
@@ -38,7 +39,7 @@ open should be called before methods like map() or join() are executed
       this must be set to transient, as flink will otherwise try to serialize it which it is not
      */
     super.open(parameters);
-    System.out.println(parameters.toString());
+    //System.out.println(parameters.toString());
```
Comment (leopetter) [inline core/src/main/java/org/bptlab/cepta/operators/WeatherLocationCorrelationFunction.java]: Gänzlich löschen oder wird es noch benötigt?

File: core/src/main/java/org/bptlab/cepta/utils/converters/PlannedTrainDataDatabaseConverter.java [hunk 4]
```diff
@@ -15,18 +17,22 @@ public PlannedTrainData fromResult(ResultSet result) throws Exception {
       event.setId(result.getInt("id"));
       event.setTrainId(result.getInt("train_id"));
       event.setLocationId(result.getInt("location_id"));
-      convertTimestamp(result.getTimestamp("planned_time"), event::setPlannedTime);
```
Comment (leopetter) [inline core/src/main/java/org/bptlab/cepta/utils/converters/PlannedTrainDataDatabaseConverter.java:18]: Werden die ganzen Methoden convert... die in dem File an vielen Stellen stehen noch benötigt?

File: core/src/test/java/org/bptlab/cepta/data_to_database/DataToDatabaseTests.java [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.sql.*;
+
+public class DataToDatabaseTests {
+
+    public void initDatabaseStuff(PostgreSQLContainer container) {
```
Comment (leopetter) [inline core/src/test/java/org/bptlab/cepta/data_to_database/DataToDatabaseTests.java]: Please change method name with stuff to something better

File: core/src/test/java/org/bptlab/cepta/data_to_database/DataToDatabaseTests.java [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+           // Create table for planned data
+           sql = createPlannedDatabaseQuery();
+           stmt.executeUpdate(sql);
+     
+        }catch(SQLException se){
+           //Handle errors for JDBC
+           se.printStackTrace();
+        }catch(Exception e){
+           //Handle errors for Class.forName
+           e.printStackTrace();
```
Comment (leopetter) [inline core/src/test/java/org/bptlab/cepta/data_to_database/DataToDatabaseTests.java:79]: combine?

File: core/src/test/java/org/bptlab/cepta/data_to_database/DataToDatabaseTests.java [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+              if(stmt!=null)
+                 conn.close();
+           }catch(SQLException se){
+           }// do nothing
+           try{
+              if(conn!=null)
+                 conn.close();
+           }catch(SQLException se){
+              se.printStackTrace();
+           }//end finally try
```
Comment (leopetter) [inline core/src/test/java/org/bptlab/cepta/data_to_database/DataToDatabaseTests.java:92]: Merge to one?

File: core/src/test/java/org/bptlab/cepta/data_to_database/DataToDatabaseTests.java [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+        }//end try
+        System.out.println("Goodbye!");
+   }
+public int checkDatabaseInput(PostgreSQLContainer container) {
```
Comment (leopetter) [inline core/src/test/java/org/bptlab/cepta/data_to_database/DataToDatabaseTests.java:97]: Basically the same remarks as in initDatabaseStuff. Maybe outsource the connection/disconnection to the database..

File: core/src/test/java/org/bptlab/cepta/live_planned_trains/LivePlannedCorrelationTests.java [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.sql.*;
+
+public class LivePlannedCorrelationTests {
+  //private PostgresConfig postgresConfig = new PostgresConfig().withHost("localhost");
+
+  public void initDatabaseStuff(PostgreSQLContainer container) {
```
Comment (leopetter) [inline core/src/test/java/org/bptlab/cepta/live_planned_trains/LivePlannedCorrelationTests.java]: Same Notes as in DataToTDatabaseTests

File: core/src/test/java/org/bptlab/cepta/weather-tests/WeatherLocationCorrelationTests.java [hunk 10]
```diff
@@ -4,65 +4,195 @@
...
+
+      // correlatedWeatherStream.print();
+      Iterator<Tuple2<WeatherData, Integer>> iterator = DataStreamUtils.collect(correlatedWeatherStream);
+      while(iterator.hasNext()){
+        Assert.assertNotEquals(4012656, (long) iterator.next().f1);
+      }
+    }    
+  }
+
+  public void initDatabaseStuff(PostgreSQLContainer container) {
```
Comment (leopetter) [inline core/src/test/java/org/bptlab/cepta/weather-tests/WeatherLocationCorrelationTests.java]: Same as said before

**Review by leopetter — APPROVED** [review]
From the perspective of a java novice it looks overall good. I just requested a few changes but nothing major. Please change them and then you can merge the PR.

File: core/src/main/java/org/bptlab/cepta/operators/DataToDatabase.java (hunk 1 above)
Comment (ma-ro) [reply core/src/main/java/org/bptlab/cepta/operators/DataToDatabase.java]: jo wird rausgenommen

File: core/src/main/java/org/bptlab/cepta/operators/DataToDatabase.java (hunk 2 above)
Comment (ma-ro) [reply core/src/main/java/org/bptlab/cepta/operators/DataToDatabase.java]: jo wird rausgenomme

File: core/src/main/java/org/bptlab/cepta/operators/WeatherLocationCorrelationFunction.java (hunk 3 above)
Comment (ma-ro) [reply core/src/main/java/org/bptlab/cepta/operators/WeatherLocationCorrelationFunction.java]: nein kann raus

File: core/src/main/java/org/bptlab/cepta/utils/converters/PlannedTrainDataDatabaseConverter.java (hunk 4 above)
Comment (ma-ro) [reply core/src/main/java/org/bptlab/cepta/utils/converters/PlannedTrainDataDatabaseConverter.java:18]: jo kommt raus

File: core/src/test/java/org/bptlab/cepta/data_to_database/DataToDatabaseTests.java (hunk 5 above)
Comment (ma-ro) [reply core/src/test/java/org/bptlab/cepta/data_to_database/DataToDatabaseTests.java]: jo

File: core/src/test/java/org/bptlab/cepta/data_to_database/DataToDatabaseTests.java (hunk 7 above)
Comment (ma-ro) [reply core/src/test/java/org/bptlab/cepta/data_to_database/DataToDatabaseTests.java:92]: add printStackTace

File: core/src/test/java/org/bptlab/cepta/data_to_database/DataToDatabaseTests.java (hunk 8 above)
Comment (ma-ro) [reply core/src/test/java/org/bptlab/cepta/data_to_database/DataToDatabaseTests.java:97]: did it!

**Pull request merged**